From patches Mon Sep 17 00:00:00 2001
From: minyard@acm.org
To: qemu-devel@nongnu.org
Cc: Corey Minyard <cminyard@mvista.com>
Date: Tue, 18 Sep 2012 15:00:28 -0500
Message-id: 1347998443-20599-2-git-send-email-minyard@acm.org
Subject: [PATCH 01/16] vl: Move init_timer_alarm() earlier

From: Corey Minyard <cminyard@mvista.com>

The timers may be used in initialization, so move them earlier so they
are ready.  A reconnect options will be added to qemu-char, and it uses
a timer, but the timer alarm wasn't initialized yet.

Signed-off-by: Corey Minyard <cminyard@mvista.com>
---
 vl.c |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/vl.c b/vl.c
index 7c577fa..1dcab37 100644
--- a/vl.c
+++ b/vl.c
@@ -3308,6 +3308,11 @@ int main(int argc, char **argv, char **envp)
         qemu_set_version(machine->hw_version);
     }
 
+    if (init_timer_alarm() < 0) {
+        fprintf(stderr, "could not initialize alarm timer\n");
+        exit(1);
+    }
+
     /* Init CPU def lists, based on config
      * - Must be called after all the qemu_read_config_file() calls
      * - Must be called before list_cpus()
@@ -3485,11 +3490,6 @@ int main(int argc, char **argv, char **envp)
 
     os_set_line_buffering();
 
-    if (init_timer_alarm() < 0) {
-        fprintf(stderr, "could not initialize alarm timer\n");
-        exit(1);
-    }
-
 #ifdef CONFIG_SPICE
     /* spice needs the timers to be initialized by this point */
     qemu_spice_init();
-- 
1.7.4.1


